### DISCUSSION TITLE
Scalacheck instances?

### DISCUSSION THREAD

**Discussion by rossabaker** [scala/scala-xml#606] [Ideas]
I needed to test an XML parser and couldn't find any scalacheck instances, so I [took matters into my own hands](https://github.com/http4s/http4s-scala-xml/pull/30/files).  These aren't specific to HTTP, so I don't think they belong in http4s in the long term.  But there's no scalacheck dependency here, so this repo doesn't feel right either.

Is there any interest, perhaps as a low profile Typelevel project?

/cc @satabin, who might find them useful in fs2-data.

**Comment by satabin** [comment]
This is indeed interesting for fs2-data, I will port it to my xml events. Thanks.

**Reply by rossabaker** [reply]
We might be creating an http4s-fs2-data.  Experimental project that everyone can depend on and get the best generators any of us have to offer: https://github.com/typelevel/scalacheck-xml

**Comment by ashawley** [comment]
FYI I tried writing a ScalaCheck test suite in #110 a while back. It has some generators to support the tests that may be of interest to that project.  It is written in the older Scala compiler code style, but it may be worth cribbing from.  It wasn't merged because of it would have added a dependency for ScalaCheck to the project; albeit it would be just for the tests not the published artifact.

**Reply by armanbilge** [reply]
I mean, doesn't scala itself depend on scalacheck?
https://github.com/scala/scala/blob/8a2cf63ee5bad8c8c054f76464de0e10226516a0/build.sbt#L40

**Reply by ashawley** [reply]
It was years ago when the semi-circular dependency between scala-xml and the compiler was a real pain when bootstrapping new 2.x compilers.

**Comment by rossabaker** [comment]
It's not perfect, but it's a start: https://github.com/typelevel/scalacheck-xml/releases/tag/v0.1.0
